### PR TITLE
Update toString/proxy.js to test stringifing functions

### PR DESCRIPTION
**Pull request by kmiller68**

### CODE REVIEW

**Comment by rwaldron** [comment]
All of this should be in a different file, maybe even more than one, but definitely not jammed into an existing test file

**Comment by leobalter** [comment]
I agree with Rick. The current file could be renamed to specify what it is currently testing. Then, we can add an individual test file for the function definitions, arrow functions, and generators.

The files can be named as `proxy-array.js`, `proxy-fn.js`, `proxy-arr-fn.js`, `proxy-gen-fn.js` or anything better.

File: test/built-ins/Object/prototype/toString/proxy.js [hunk 1]
```diff
@@ -38,3 +47,50 @@ assert.sameValue(
...
+assert.sameValue(
+  Object.prototype.toString.call(generatorProxy), '[object GeneratorFunction]', 'generator function proxy'
+);
+assert.sameValue(
+  Object.prototype.toString.call(generatorProxyProxy),
+  '[object GeneratorFunction]',
+  'proxy for generator function proxy'
+);
+
+delete generatorProxy.__proto__[Symbol.toStringTag];
```
Comment (leobalter) [inline test/built-ins/Object/prototype/toString/proxy.js]: is this really necessary? If so, it's probably interesting to have one another test file for it and avoid using `__proto__`.

Comment (kmiller68) [inline test/built-ins/Object/prototype/toString/proxy.js]: Are you referring to testing generators as a whole? Or are you more referring to testing generators' Symbol.toStringTag?

Comment (kmiller68) [inline test/built-ins/Object/prototype/toString/proxy.js]: I could do `generatorProxy.constructor.prototype` or `(function*() {}).constructor.prototype`. Would that be preferable? I'm not sure how much value there is in assuming both the `constructor` and `prototype` properties work correctly over assuming `__proto__` works correctly, though.

Comment (leobalter) [inline test/built-ins/Object/prototype/toString/proxy.js]: you already have `generatorProxy.constructor.prototype` so I think that would be the best option.

**Comment by rwaldron** [comment]
@kmiller68 ping

**Comment by leobalter** [comment]
@kmiller68 Please tell me if you need any support to fix this PR, technically or even with available time.

**Pull request merged**